impl/client/src/main/javascript/web/prompting/builders/TextInputBuilder.js

```diff
@@ -49,76 +49,110 @@
  * @class
  * @extends FormattedParameterWidgetBuilderBase
  */
-define(["common-ui/util/util", 'dojo/number', 'cdf/components/TextInputComponent', './FormattedParameterWidgetBuilderBase', 'common-ui/jquery-clean'],
-  function (Util, DojoNumber, TextInputComponent, FormattedParameterWidgetBuilderBase, $) {
+define([
+  "common-ui/util/util",
+  "dojo/number",
+  "cdf/components/TextInputComponent",
+  "./FormattedParameterWidgetBuilderBase",
+  "common-ui/jquery-clean",
+  "common-ui/underscore"
+], function(Util, DojoNumber, TextInputComponent, FormattedParameterWidgetBuilderBase, $, _) {
 
-    return FormattedParameterWidgetBuilderBase.extend({
-      /**
-       * Builds the widget and returns a TextInputComponent
-       * @method
-       * @name TextInputBuilder#build
-       * @param {Object} args The arguments to build the widget in accordance with [the CDF documentation]{@link http://localhost:8080/pentaho/api/repos/:public:plugin-samples:pentaho-cdf:pentaho-cdf-require:30-documentation:30-component_reference:10-core:37-TextInputComponent:text_input_component.xcdf/generatedContent}.
-       * @param {PromptPanel} args.promptPanel - The instance of PromptPanel
-       * @param {ParameterDefinition} args.promptPanel.paramDefn - The parameter definition
-       * @param {Parameter} args.param - The Parameter instance
-       * @returns {TextInputComponent} The TextInputComponent built
-       */
-      build: function (args) {
-        function parseNumber(val){
-          try{
-            return DojoNumber.parse(val, { locale : Util.normalizeDojoLocale(SESSION_LOCALE) });
-          } catch(e) {
-            return DojoNumber.parse(val, { locale : "en" });
-          }
+  var currentDashboard;
+
+  return FormattedParameterWidgetBuilderBase.extend({
+    /**
+     * Builds the widget and returns a TextInputComponent
+     * @method
+     * @name TextInputBuilder#build
+     * @param {Object} args The arguments to build the widget in accordance with [the CDF documentation]{@link http://localhost:8080/pentaho/api/repos/:public:plugin-samples:pentaho-cdf:pentaho-cdf-require:30-documentation:30-component_reference:10-core:37-TextInputComponent:text_input_component.xcdf/generatedContent}.
+     * @param {PromptPanel} args.promptPanel - The instance of PromptPanel
+     * @param {ParameterDefinition} args.promptPanel.paramDefn - The parameter definition
+     * @param {Parameter} args.param - The Parameter instance
+     * @param {Boolean?} args.param.attributes.addClearIcon - The whether creating a clear icon for the text input
```

Comment:
Maybe indicate default value on jsdoc

`@param {Boolean} [args.param.attributes.addClearIcon=false] - Whether a clear icon should be created for the text input component.`